### ISSUE TITLE
[REVIEW]: Manif: A micro Lie theory library for state-estimation in robotics applications

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1371]
**Submitting author:** @artivis (<a href="http://orcid.org/0000-0001-5279-8251">Jeremie Deray</a>)
**Repository:** <a href="https://github.com/artivis/manif" target ="_blank">https://github.com/artivis/manif</a>
**Version:** 0.0.2
**Editor:** @poulson
**Reviewer:** @jordigh, @drvinceknight, @alex-konovalov
**Archive:** <a href="https://doi.org/10.5281/zenodo.3626651" target="_blank">10.5281/zenodo.3626651</a>

## Status

[![status](http://joss.theoj.org/papers/e3fc778689407f0edd19df8c2089c160/status.svg)](http://joss.theoj.org/papers/e3fc778689407f0edd19df8c2089c160)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/e3fc778689407f0edd19df8c2089c160"><img src="http://joss.theoj.org/papers/e3fc778689407f0edd19df8c2089c160/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/e3fc778689407f0edd19df8c2089c160/status.svg)](http://joss.theoj.org/papers/e3fc778689407f0edd19df8c2089c160)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/artivis/manif" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@jordigh & @drvinceknight & @alex-konovalov, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @poulson know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/artivis/manif">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Version:** 0.0.2
- [x] **Authorship:** Has the submitting author (@artivis) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/artivis/manif">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Version:** 0.0.2
- [x] **Authorship:** Has the submitting author (@artivis) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/artivis/manif">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Version:** 0.0.2
- [x] **Authorship:** Has the submitting author (@artivis) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @jordigh, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1371 with the following error: 

 Can't find any papers to compile :-(

**Comment by artivis** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01371/joss.01371/10.21105.joss.01371.pdf)

**Comment by alex-konovalov** [comment]
Sorry for delay with reviewing, it may have to wait till June now, but completing it in June looks reliably doable.

**Comment by drvinceknight** [comment]
I have started my review, and everything looks good apart from minor comments that I've left here: https://github.com/artivis/manif/pull/79, I need to work through my remaining tick boxes (all regarding functionality).

**Comment by danielskatz** [comment]
👋 this is just a ping to those involved in this paper (author, reviewers, editor) that nothing has happened in this review thread in about a month.  It looks like @alex-konovalov hasn't started, but says he will do this in June. @drvinceknight has done a large part of his review, but needs to finish.  @jordigh hasn't started their review.

**Comment by artivis** [comment]
Hi, thanks for the ping. We've addressed @drvinceknight remarks already. I will update the branch were the paper lives to better reflect those changes.

**Comment by drvinceknight** [comment]
I plan on completing my review on Friday.

**Comment by alex-konovalov** [comment]
Still the plan to finish it in June.

**Comment by drvinceknight** [comment]
I have completed my review: one request would be to include OS requirements in the installation requirements, I believe that this is only supported on linux (because of the Eigen 3) dependency (I could be mistaken).

**Comment by artivis** [comment]
@drvinceknight Thanks for the review.
There isn't really an OS requirement, `manif` works on linux & OSX, we did not try to compile it with Windows. That's something that will land on the todo-list.  
Notice that Eigen3 is available for Windows as well.

**Comment by drvinceknight** [comment]
Ok, I failed to install Eigen3 on OSX, could you point me at the instructions or perhaps include the install instructions in `manif`'s documentation?

**Comment by artivis** [comment]
I do not have an OSX machine at the moment but as I recall there should be a `brew install eigen3` -like command. We will add OSX instructions to `manif`'s doc once we can confirm them.

**Comment by alex-konovalov** [comment]
@drvinceknight @artivis I have competed my review. `brew install eigen` worked for me (`brew install eigen3` is not correct since there is no `eigen3` formula). I have left a couple of suggestions at https://github.com/artivis/manif/issues/90. The only tickboxes left empty are about version number (because none so far) and about performance (it seems there are no claims about it, so I've left it empty).

**Comment by artivis** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01371/joss.01371/10.21105.joss.01371.pdf)

**Comment by artivis** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01371/joss.01371/10.21105.joss.01371.pdf)

**Comment by artivis** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01371/joss.01371/10.21105.joss.01371.pdf)

**Comment by artivis** [comment]
@alex-konovalov @drvinceknight we addressed most @alex-konovalov's comments (https://github.com/artivis/manif/issues/90). Remains only releasing the library which we will do in the upcoming days after a couple small fixes are merged to the main branch. We will then rebase the branch associated to this paper and let you know.

ps: sorry for spamming whedon, I do not know how to compile this paper locally :s

**Comment by labarba** [comment]
@artivis — could we have a status update on your revision? 

@poulson — one reviewer (@jordigh) has all items unchecked still; have you given up on their review?

**Comment by poulson** [comment]
:wave: @labarba: Good question, and I will personally email @jordigh now in hopes of increasing the chances of a response.

**Comment by artivis** [comment]
@labarba Sorry for the late answer.
We are done with the revision. The last thing remaining is to generate a release (and tag) to associate to the paper. However we'd like to do that at the very end of the review process in case some other fixes are required.

**Comment by jordigh** [comment]
Some initial comments, will work on this more later:

https://github.com/artivis/manif/issues/98
https://github.com/artivis/manif/issues/99
https://github.com/artivis/manif/pull/100

**Comment by poulson** [comment]
Thank you for taking the time to start working on this review, Jordi (@jordigh)!

**Comment by poulson** [comment]
Hi Jordi (@jordigh): I hate to ping you, but the other two reviewers are essentially done and the process is currently blocked on you finishing your review.

Do you think you would be able to finish in the next few days?

**Comment by kthyng** [comment]
Another ping here for @jordigh — will you be able to complete your review soon?

**Comment by poulson** [comment]
FWIW, via email @jordigh stated that he would have completed over last weekend.

**Comment by danielskatz** [comment]
👋 @jordigh - any update on the status of your review?
@poulson - if @jordigh doesn't reply soon, would you be comfortable proceeding with the other 2 reviews that are complete?

**Comment by poulson** [comment]
Hi @danielskatz -- yes, I suggest we move on without @jordigh's review. I am in a similar deadlock on another paper with them.

**Comment by poulson** [comment]
Hi @alex-konovalov -- I was curious if you could clarify what your current concerns are with respect to the functionality of the library.

**Comment by jordigh** [comment]
In case you still want my review, I have some lingering concerns,

https://github.com/artivis/manif/issues/107

https://github.com/artivis/manif/issues/106

Everything else seems fine. No DOIs for the citations in the paper, but I think everything cited is just other libraries that don't have DOIs or preprints that haven't been published yet.

**Comment by poulson** [comment]
Thank you, @jordigh!!!

@alex-konovalov :wave: -- would you mind clarifying what your remaining functionality concerns are? We are nearing the end of the review process.

**Comment by alex-konovalov** [comment]
@poulson the authors so far addressed (thanks) all my suggestion from https://github.com/artivis/manif/issues/90 except the one about the proper release with a version number allocated - they wrote there that it will be released soon.

**Comment by artivis** [comment]
@poulson we addressed all remarks from artivis/manif#90 but the release with a version number. We would like to do so once all other reviews are addressed as a last step accompanying the publication of this paper. If this is any trouble please let me know.
The final remark we have to address is this of @jordigh at artivis/manif#107 which we are investigating now.

**Comment by poulson** [comment]
Thank you for the updates, @alex-konovalov and @artivis! I look forward to the resolution of @jordigh's last issue and will do a final pass myself after that for things like typos and DOIs.

**Comment by artivis** [comment]
@poulson We just addressed artivis/manif#107. We thus are forward to your final pass!

**Comment by danielskatz** [comment]
👋 @poulson - it looks like this is waiting for you

**Comment by poulson** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1371 with the following error: 

 Can't find any papers to compile :-(

**Comment by poulson** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01371/joss.01371/10.21105.joss.01371.pdf)

**Comment by poulson** [comment]
Hi @artivis -- I just finished giving the PDF a careful read and have the following minor comments:

1. The opening paragraph could use a few more citations, such as for one or two authors part of the 'remarkable effort', and for either a classical mathematical text on smooth manifolds (e.g., John Lee's "Introduction to Smooth Manifolds") or an application-specific one, e.g., your own publication that you cite in the "Mathematical foundations of the library..." paragraph, which I would recommend deleting in favor of absorbing the citation at the end of the initial sentence introducing the term 'manifold'.

2. I recommend rewording the sentence "The manif library has been developed to make easily accessible the most common operations on Lie groups in state estimation." to "The manif library provides simple interfaces to the most common operations on Lie groups in state estimation."

3. I recommend rewording "Its design is similar to Eigen, so that all Lie group classes defined in have common that they inherit from a templated base class using static polymorphism." to "Its design is similar to Eigen, in that all Lie group classes inherit from a templated base class using static polymorphism."

4. I recommend rewording "This allows for the possibility the write generic code without paying the price of pointer arithmetic." to "This allows for writing generic code without paying the price of pointer arithmetic."

5. It would be useful to define the Special Orthogonal (SO(n)) and Special Euclidean (SE(n)) groups by name given that you are introducing the term manifold. While we're at it, it would be useful to provide a reference for 'Lie groups' upon their initial introduction.

6. I recommend rewording the sentence "Sophus C++ implementation..." to "Sophus is a C++ implementation...".

7. I recommend rewording the sentence "wave geometry Manifold geometry with fast..." to "wave_geometry is a library for manifold geometry with fast...".

8. I recommend adding a comma after "C++11" in the sentence "Our work differs from wave_geometry in that it relies on C++11 which is...".

**Comment by poulson** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- None

MISSING DOIs

- https://doi.org/10.1109/crv.2018.00027 may be missing for title: Manifold Geometry with Fast Automatic Derivatives and Coordinate Frame Semantics Checking in C++
- https://doi.org/10.1007/978-1-4419-7719-9_6 may be missing for title: Boost C++ libraries

INVALID DOIs

- None
```

**Comment by poulson** [comment]
Hi @artivis -- would you mind fixing the above two missing DOIs in addition to adding the requested references in my above (lengthy) comment?

**Comment by artivis** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01371/joss.01371/10.21105.joss.01371.pdf)

**Comment by artivis** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1109/CRV.2018.00027 is OK
- 10.1017/9781316671528 is OK

MISSING DOIs

- https://doi.org/10.1007/978-1-4419-7719-9_6 may be missing for title: Boost C++ libraries

INVALID DOIs

- None
```

**Comment by artivis** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01371/joss.01371/10.21105.joss.01371.pdf)

**Comment by artivis** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01371/joss.01371/10.21105.joss.01371.pdf)

**Comment by artivis** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1109/CRV.2018.00027 is OK
- 10.1007/978-1-4419-7719-9_6 is OK
- 10.1017/9781316671528 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by artivis** [comment]
Hi @poulson, sorry for the late reply, end of the year etc...
I addressed your comments as follows:

1. Added two relevant references for Lie theory formulation in robotics.
Added a classical mathematical reference at the very end of the first paragraph.
Kept the reference to our own publication as is.

2/3/4/6/7/8. Reworded sentences as suggested.

5. Introduced SO/SE by name. Not sure what you meant by
"it would be useful to provide a reference for 'Lie groups' upon their initial introduction."

Moreover the missing DOIs were fixed.

Looking forward to further comments.

**Comment by poulson** [comment]
Hi @artivis -- the rewrite looks wonderful to me! Thank you.

Would you mind creating a DOI on zenodo and posting the link here?

**Comment by artivis** [comment]
@poulson Sure thing, just the time for me to check how to do so.

**Comment by artivis** [comment]
@poulson So I found a 'Zenodo how to'  that I'm following. To create a DOI I have first to issue a release of the repo. If the document is done with reviews and if you agree I propose the following course of action:
1. I'll merge the branch `joss` on which the paper lives to the main branch (`devel`) and then sync the release branch (`master`).
2. Issue a first tagged release from the release branch (`master`)
3. Create a DOI on Zenodo.

Is that ok with you?

**Comment by poulson** [comment]
@artivis I believe you will need to use the same version from this paper. Is there something wrong with using the `joss` branch?

**Comment by artivis** [comment]
@poulson The `joss` branch was forked a year ago now. If the functionalities and API haven't change much, the library has known a lot of internal changes, bug-fixes and such. In my mind this branch was always meant to be merged at some point, especially because the repo is supposed to follow a develop/release branching scheme. But maybe the workflow is different here?

**Comment by poulson** [comment]
@alex-konovalov @drvinceknight @jordigh: Which branch of the software did you review? The `joss` branch or `master`?

**Comment by artivis** [comment]
> @alex-konovalov @drvinceknight @jordigh: Which branch of the software did you review? The `joss` branch or `master`?

** `joss` branch or `devel`.  
Notice that all changes requested by the reviewers (except those concerning the paper itself) were done in branch `devel`.

**Comment by alex-konovalov** [comment]
@poulson I can't recollect, but I think my suggestions did not depend much on the branch...

**Comment by poulson** [comment]
In that case, @artivis, can you post the DOI with your commended process? I think it makes the most sense to use the version that the reviewers reviewed.

We should be able to quickly move onto finalizing the paper and formally recommending it for acceptance.

Thank you for your patience!

**Comment by artivis** [comment]
Dear @poulson, 
I finally published a release of the library on Zenodo following the process previously proposed, you may find it [here](https://zenodo.org/record/3634976). The DOI associated to the release is: **10.5281/zenodo.3634976**, while the _global_ DOI associated to any version is: **10.5281/zenodo.3626651**. As I'm not sure which one you require I post both.

Please let me know if there is anything else to do on my side,
Thanks.

**Comment by poulson** [comment]
@whedon set 10.5281/zenodo.3626651 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3626651" target="_blank">10.5281/zenodo.3626651</a> is the archive.

**Comment by poulson** [comment]
@whedon set "branch: devel" as version

**Comment by whedon** [comment]
OK. "branch: devel" is the version.

**Comment by poulson** [comment]
@whedon set 0.0.2 as version

**Comment by whedon** [comment]
OK. 0.0.2 is the version.

**Comment by poulson** [comment]
:wave: @openjournals/joss-eics, this paper should be ready for final processing!

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1109/CRV.2018.00027 is OK
- 10.1007/978-1-4419-7719-9_6 is OK
- 10.1017/9781316671528 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1309

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1309, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
👋 @arfon - some of the refs in the XML have some new problems, and one has an old problem

**Comment by danielskatz** [comment]
👋 @artivis - please merge https://github.com/artivis/manif/pull/125 and https://github.com/artivis/manif/pull/126

**Comment by danielskatz** [comment]
@arfon - note https://github.com/artivis/manif/pull/126 which might fix some of the xml issues

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1109/CRV.2018.00027 is OK
- 10.1007/978-1-4419-7719-9_6 is OK
- 10.1017/9781316671528 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1317

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1317, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
Sorry, I found a couple more things I missed - can you merge https://github.com/artivis/manif/pull/128 as well?

**Comment by artivis** [comment]
> Sorry, I found a couple more things I missed - can you merge [artivis/manif#128](https://github.com/artivis/manif/pull/128) as well?

@danielskatz Done :+1:

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1109/CRV.2018.00027 is OK
- 10.1007/978-1-4419-7719-9_6 is OK
- 10.1017/9781316671528 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1318

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1318, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1229844854840020993) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1319
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01371](https://doi.org/10.21105/joss.01371)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by danielskatz** [comment]
Thanks to @jordigh & @drvinceknight & @alex-konovalov for reviewing, and @poulson for editing!

Congratulations to @artivis & @joansola!

**Comment by danielskatz** [comment]
👋 @arfon - can you take a look at the XML file and see if it seems ok?  If so, please go ahead and close this issue.  But in either case, I think this has demonstrated that the way we extract info from .bib files into the XML file needs work, outside this specific issue.

**Comment by artivis** [comment]
Thanks all for the reviews and help through this process!

**Comment by arfon** [comment]
> 👋 @arfon - can you take a look at the XML file and see if it seems ok? If so, please go ahead and close this issue.

:+1: things look good to me. 

> But in either case, I think this has demonstrated that the way we extract info from .bib files into the XML file needs work, outside this specific issue.

Would it be possible to open a tracking issue on the JOSS repo for this outlining the general issue?

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01371/status.svg)](https://doi.org/10.21105/joss.01371)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01371">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01371/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01371/status.svg
   :target: https://doi.org/10.21105/joss.01371
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01371/status.svg)](https://doi.org/10.21105/joss.01371)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
